src/Widgets/WorkspaceInsertThumb.vala

```diff
@@ -43,8 +44,8 @@ namespace Gala
 			x_align = Clutter.ActorAlign.CENTER;
 
 			var drop = new DragDropAction (DragDropActionType.DESTINATION, "multitaskingview-window");
-			drop.crossed.connect ((hovered) => {
-				if (!Prefs.get_dynamic_workspaces ())
+			drop.crossed.connect ((target, hovered) => {
+				if (!Prefs.get_dynamic_workspaces () && (target != null && target is WindowClone))
```

Comment:
This behaviour is handled directly in `WindowClone`: https://github.com/elementary/gala/blob/ca32785fe4ffc64eaa01d9c87cbcb6b4336bca6a/src/Widgets/WindowClone.vala#L605

So instead of handling it here, `WindowClone` watches for crossed destinations and calls `set_window_thumb` which resizes the `WorkspaceInsertThumb` and adds a pulse animation.